core/services/gas/optimism_estimator.go

```diff
@@ -0,0 +1,193 @@
+	"time"
+
+	optimismfees "github.com/ethereum-optimism/go-optimistic-ethereum-utils/fees"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/core/utils"
+	"github.com/tidwall/gjson"
+	"go.uber.org/multierr"
+)
+
+// It's always 0.015 GWei
+// See: https://www.notion.so/How-to-pay-Fees-in-Optimistic-Ethereum-f706f4e5b13e460fa5671af48ce9a695
+const optimisml1GasPrice = 15000000000
+
+var _ Estimator = &optimismEstimator{}
+
+//go:generate mockery --name optimismRPCClient --output ./mocks/ --case=underscore --structname OptimismRPCClient
+type optimismRPCClient interface {
+	Call(result interface{}, method string, args ...interface{}) error
+}
+
+// TODO: Separate tracker/estimator? can it be done?
+type optimismEstimator struct {
+	utils.StartStopOnce
+
+	config     Config
+	client     optimismRPCClient
+	pollPeriod time.Duration
+
+	gasPriceMu sync.RWMutex
+	l1GasPrice *big.Int
+	l2GasPrice *big.Int
+
+	chForceRefetch chan (chan struct{})
+	chInitialised  chan struct{}
+	chStop         chan struct{}
+	chDone         chan struct{}
+}
+
+// NewOptimismEstimator returns a new optimism estimator
+func NewOptimismEstimator(config Config, client optimismRPCClient) Estimator {
+	return &optimismEstimator{
+		utils.StartStopOnce{},
+		config,
+		client,
+		10 * time.Second,
+		sync.RWMutex{},
+		nil,
+		nil,
+		make(chan (chan struct{})),
+		make(chan struct{}),
+		make(chan struct{}),
+		make(chan struct{}),
+	}
+}
+
+func (o *optimismEstimator) Start() error {
+	return o.StartOnce("OptimismEstimator", func() error {
+		go o.run()
+		<-o.chInitialised
+		return nil
+	})
+}
+func (o *optimismEstimator) Close() error {
+	return o.StopOnce("OptimismEstimator", func() error {
+		close(o.chStop)
+		<-o.chDone
+		return nil
+	})
+}
+
+func (o *optimismEstimator) run() {
+	defer close(o.chDone)
+
+	t := o.refreshPrices()
+	close(o.chInitialised)
+
+	for {
+		select {
+		case <-o.chStop:
+			return
+		case ch := <-o.chForceRefetch:
+			t.Stop()
+			t = o.refreshPrices()
+			close(ch)
+		case <-t.C:
+			t = o.refreshPrices()
+		}
+	}
+}
+
+// OptimismGasPricesResponse is the shape of the response when calling rollup_gasPrices
+type OptimismGasPricesResponse struct {
+	L1GasPrice *big.Int
+	L2GasPrice *big.Int
+}
+
+func (g *OptimismGasPricesResponse) UnmarshalJSON(b []byte) error {
+	var l1Hex string = gjson.GetBytes(b, "l1GasPrice").Str
+	var l2Hex string = gjson.GetBytes(b, "l2GasPrice").Str
+	var l1 *hexutil.Big
+	var l2 *hexutil.Big
+	if err := multierr.Combine(l1.UnmarshalText([]byte(l1Hex)), l2.UnmarshalText([]byte(l2Hex))); err != nil {
+		return err
+	}
+	g.L1GasPrice = l1.ToInt()
+	g.L2GasPrice = l2.ToInt()
+	return nil
+}
+
+func (o *optimismEstimator) refreshPrices() (t *time.Timer) {
+	var res OptimismGasPricesResponse
+	t = time.NewTimer(utils.WithJitter(o.pollPeriod))
```

Comment:
hm something seems to be missing with the usage of this timer - it's just returned from this method, but then does not seem to be actually used outside either?

Reply:
Yup, it is used in `run()`

Reply:
ah yes `case <-t.C:`